staging/src/k8s.io/apimachinery/pkg/util/net/interface.go

```diff
@@ -244,23 +244,22 @@ func getMatchingGlobalIP(addrs []net.Addr, family AddressFamily) (net.IP, error)
 
 // getIPFromInterface gets the IPs on an interface and returns a global unicast address, if any. The
 // interface must be up, the IP must in the family requested, and the IP must be a global unicast address.
-func getIPFromInterface(intfName string, forFamily AddressFamily, nw networkInterfacer) (net.IP, error) {
-	intf, err := nw.InterfaceByName(intfName)
-	if err != nil {
-		return nil, err
+func getIPFromInterface(intf *net.Interface, forFamily AddressFamily, nw networkInterfacer) (net.IP, error) {
```

Comment:
I don't follow why this change was necessary.  It looks like the core change is the `chooseHostInterface` to be per OS.

Reply:
I made this change to reduce repeated code.
I needed to use the same logic to find a valid IP, but unlike the Linux routing table which has the name of the interface, the Windows routing table contains the index of the interface.

I've just pushed changes which have reduced the delta of the changes I made, but I am still touching that function.